### ISSUE TITLE
`getimagesizefromstring` Error 

### ISSUE THREAD

**Issue by mendelB** [oscarotero/Embed#249]
First off - thanks for an awesome project!!

I've bumped into an issue where attempting to call `$info->providerIcon` from my Embed object resulted in an `getimagesizefromstring(): read error!`

The URL fed to the `Embed` constructor was `https://www.vox.com`. 

e.g.

```
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Embed\Embed;

class EmbedController extends Controller
{
    function index()
    {
	$info = Embed::create('https://www.vox.com');

    	return response()->json([
    		'icon' => $info->providerIcon,
    	]);
    }
}
```

![image](https://user-images.githubusercontent.com/12417657/32734296-a2b3c1cc-c85f-11e7-96ff-6d3db7c9f398.png)

What seems to be happening is the call to `https://www.vox.com/favicon.ico` (which is always [made](https://github.com/oscarotero/Embed/blob/master/src/Adapters/Adapter.php#L354), is returning a `404` with plain text type content of "Not Found".  (in contrast with most sites which would generally return some HTML)

When this string is given as an argument to the `CurlDispatcher`s [getimagesizefromstring](https://github.com/oscarotero/Embed/blob/master/src/Http/CurlDispatcher.php#L185) method, it errors out for some reason. 

(Oddly enough it doesn't error out if the string contains HTML of some sort (Which I imagine is why this error doesn't occur with most sites that actually return some richer `text/html` response content)
![image](https://user-images.githubusercontent.com/12417657/32734500-35226a72-c860-11e7-8b27-74b77d7697a0.png)
, which is probably why this doesn't break ordinarily for other sites that 404 from `/favicon.ico`, it only seems to freak out when it's a plain string, which in the case of Vox - it is).

This error wasn't occurring in another project where we were using `Embed`. And I believe this is because that project was using an older version that did not contain [these changes](https://github.com/oscarotero/Embed/commit/6a4edf4553780e010c867be4db9fdfd9356e951f#diff-fe01a28113bef9643bb3da300d890072) (ref #242) which cause the `onBody` function to `return` when the data type was off, before attempting to call the `getimagefromstring`.

Thanks very much in advance for your time! Please let me know if there's any way I can help with a fix for this admittedly somewhat edge case.

**Comment by oscarotero** [comment]
Yes, that error should be silenced, because `getimagefromstring` is used not only to get the image size but also to check whether the response is a valid image or not.
I just released a new version with this bug fixed. Thanks for reporting.

**Comment by mendelB** [comment]
Thanks so much @oscarotero!

**Comment by byndstd** [comment]
sorry to re open this issue but I tried with a link and I get this error : getimagesizefromstring(): Read error!
 in my web and also in your demo web; link of image is your demo website with this error

https://image.ibb.co/cxrAoK/Untitl_1.png

**Comment by oscarotero** [comment]
Thanks for notify this. It's fixed here 87aa0ed9adde31a4b31ec5df4d6424bf9c8aa0d4
You can use the dev-master version until a new tag is released.
